pages/quizzes.tsx

```diff
@@ -21,19 +21,19 @@ export default function Quizzes({
   const [chosenTopics, setChosenTopics] = useState<string[]>([]);
   const [filteredQuizzes, setFilteredQuizzes] = useState<ChinguQuiz.Quiz[]>([]);
 
-  const subjectFilterCallback = (quizTagArray: string[]) => {
-    let totalMatches = 0;
-    for (let i = 0; i < quizTagArray.length; i += 1) {
-      if (chosenTopics.indexOf(quizTagArray[i]) >= 0) {
-        totalMatches += 1;
-      }
-    }
-    // Only includes quiz if all quiz tag are matched to topic selection in the UI
-    return totalMatches === quizTagArray.length;
-  };
-
   // Handle filtering of quizzes
   useEffect(() => {
+    const subjectFilterCallback = (quizTagArray: string[]) => {
```

Comment:
Same for this - Doesn't seem like it's needed to update Next.

Reply:
This was also a warning but the fix was so obvious and simple and seemed appropriate to include in this PR

Reply:
Hmm, it's a weird warning. I don't really see why it would complain about this. To me, it doesn't make much difference whether or not this is inside or outside `useEffect`.

@ZumDeWald Any idea about this one?

Reply:
This is a static function that is used by the `useEffect` it is now inside... that makes no sense. There is no need to re-instate the static function every time the effect runs.

It is using `chosenTopics` which is in state, but still should not have to be inside an effect for any reason I can think up.

@jaytula , was husky failing this because of TSC or ESLint?

Reply:
> This is a static function that is used by the `useEffect` it is now inside... that makes no sense. There is no need to re-instate the static function every time the effect runs.
> 
> It is using `chosenTopics` which is in state, but still should not have to be inside an effect for any reason I can think up.
> 
> @jaytula , was husky failing this because of TSC or ESLint?

Maybe I'm mistaken but in what I understand about the React Component Life Cycle, the `subjectFilterCallback` would be recreated every time and thus be a different function causing `useEffect` to run so it it either has to be in the `useEffect` or wrapped in a `useCallback`.

Sorry, I don't know much about the husky setup but I'm guessing the failures are not being captured there but moreso on the deploy at Vercel.

Reply:
Either way, since it is only used by the effect it certainly does not hurt to have it in there.
I could be wrong (been a while since I was deep in React) but I can't remember any reason a static function used by an effet has to be declared inside of it (even if it is referring to state inside the function).

If this was an issue it seems like React, ESLint, or TSC would have complained before and made us fix it.